dask/dataframe/io/parquet.py

```diff
@@ -517,20 +542,78 @@ def _read_pyarrow(fs, fs_token, paths, columns=None, filters=None,
 
     all_columns = index_names + column_names
 
+    # Find non-empty pieces
+    non_empty_pieces = []
+    # Determine valid pieces
+    for piece in dataset.pieces:
+        pf = piece.get_metadata(pq.ParquetFile)
+        # non_empty_pieces.append(piece)
+        if pf.num_row_groups > 0:
+            non_empty_pieces.append(piece)
+
+    # Sort pieces naturally
+    non_empty_pieces = _nat_sorted_pieces(non_empty_pieces)
+
+    # Determine divisions
+    if non_empty_pieces and index_names:
+        # Try to find parquet column index of divisions column
+        divisions_col_index = None
+        if len(index_names) == 1 and pandas_metadata and pandas_metadata['columns']:
+            col_names = [e['name'] for e in pandas_metadata['columns']]
+            try:
+                divisions_col_index = col_names.index(index_names[0])
+            except ValueError:
+                pass
+
+        if divisions_col_index is not None:
+            # Compute min/max for column in each row group
+            min_maxs = []
+            last_max = None
+            for piece in non_empty_pieces:
+                pf = piece.get_metadata(pq.ParquetFile)
+                rg = pf.row_group(0)
+                col_meta = rg.column(divisions_col_index)
+                stats = col_meta.statistics
+                if last_max is None or last_max < stats.min:
+                    min_maxs.append((stats.min, stats.max))
+                    last_max = stats.max
+                else:
+                    # Divisions not valid
+                    min_maxs = None
+                    break
+
+            if min_maxs:
+                divisions = [mn for mn, mx in min_maxs] + [min_maxs[-1][1]]
+            else:
+                divisions = (None,) * (len(non_empty_pieces) + 1)
+        else:
+            divisions = (None,) * (len(non_empty_pieces) + 1)
+    elif non_empty_pieces:
+        divisions = (None,) * (len(non_empty_pieces) + 1)
+    else:
+        divisions = (None, None)
+
+    # Build task
     dtypes = _get_pyarrow_dtypes(schema, categories)
     dtypes = {storage_name_mapping.get(k, k): v for k, v in dtypes.items()}
 
     meta = _meta_from_dtypes(all_columns, dtypes, index_names, column_index_names)
     meta = clear_known_categories(meta, cols=categories)
 
+    # Handle converting divisions to timestamps
+    if divisions[0] is not None and meta.index.dtype.kind == 'M':
+        # Why is this factor of 1000 needed???
```

Comment:
This could be also a factor of 1M or 1 depending on the precision of the time. Pandas uses nanoseconds by default whereas we use the Parquet preferred default of microsecond precision in Parquet. But as Parquet supports many precisions and Pandas only a single one, you will have to look at the actual Parquet schema to infer the scaling factor, the `meta` object will not have sufficient information to determine it.

Reply:
Thanks for the help on this one!